shell/app-shell/app-shell.js

```diff
@@ -246,6 +257,16 @@ class AppShell extends Xen.Debug(Xen.Base, log) {
       state.showhint = Boolean(state.suggestions && state.suggestions.length > 0);
     }
   }
+  _updateSerialization(state) {
+    // TODO(sjmiles): wait 4s for context :(
```

Comment:
What is the longer term approach here, should we file an issue to discuss with @shans?

Reply:
Ideally the context logic sends a signal when context meets some heuristic for stable. I temporarily just skipped that work in favor of the timeout.